Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -110,6 +110,25 @@ private extension CardPresentPaymentStore {
             // https://stripe.dev/stripe-terminal-ios/docs/Enums/SCPPaymentIntentStatus.html#/c:@E@SCPPaymentIntentStatus@SCPPaymentIntentStatusRequiresCapture
             // TODO. Persist PaymentIntent, so that we can use it later to print a receipt
             // Deferred to https://github.com/woocommerce/woocommerce-ios/issues/3825
+
+            // Note: Here we transition from the Stripe Terminal Payment Intent
+            // to the WCPay backend managed Payment Intent, which we need
+            // to use to capture the payment.
+
+            self.remote.captureOrderPayment(for: siteID, orderID: orderID, paymentIntentID: intent.id) { result in
```

Comment:
Not a blocker, but Would it make sense to extract this to a separate private method? It would help us avoid nesting this call into the receiveValue block.

Reply:
So... I was thinking about this a bit and it reminded me of something I considered yesterday too.... there is a nice chaining of `createPaymentIntent`, `collectPaymentMethod`, `processPayment` already in place in `StripeCardReaderService` `capturePayment` where it might make sense to add intent capture as well. However, we’d have to do something similar for what we did for tokens since we don’t want to introduce a dependency between `StripeCardReaderService` and Networking. We’d need a “PaymentIntentCaptureProvider” or some such.

So, if I were going to move the call to self.remote.captureOrderPayment to its own private function, I’d be inclined to consider going the whole way to moving it and adding it to that chain. What do you think? I can open an issue for whichever path we choose (and commit this as it is for now)

Reply:
I have been thinking about this on my side too and I think that, if we changed it, we should go the other way around. 

I logged a ticket earlier today for adding a new store to manage the business logic related to WCPay (#4011 ) I think we might need to consider to pull this to that store. 

Maybe we can chat about this at another time? 😊

Reply:
Sounds good.